analytics/integration_test/src/integration_test.cc

```diff
@@ -341,4 +341,50 @@ TEST_F(FirebaseAnalyticsTest, TestSetConsent) {
   did_test_setconsent_ = true;
 }
 
+// Test that it compiles and runs on all platforms.
+TEST_F(FirebaseAnalyticsTest, TestSetDefaultEventParameters) {
+  LogInfo("Testing SetDefaultEventParameters().");
+
+  // Set some default parameters.
+  std::map<std::string, firebase::Variant> defaults;
+  defaults["default_int"] = 123;
+  defaults["default_string"] = "default_value";
+  firebase::analytics::SetDefaultEventParameters(defaults);
+
+  // Log an event - the defaults should be included automatically by the
+  // underlying SDK if logging immediately after setting is supported.
+  // Verification might need manual checking in the Analytics console or
+  // via platform-specific debug logs if possible.
+  firebase::analytics::LogEvent("event_with_defaults");
+  LogInfo("Logged event_with_defaults");
+
+  // Clear one default parameter by setting it to null.
+  defaults["default_int"] = firebase::Variant::Null();
+  firebase::analytics::SetDefaultEventParameters(defaults);
+
+  // Log another event.
+  firebase::analytics::LogEvent("event_with_one_default_cleared");
+  LogInfo("Logged event_with_one_default_cleared");
+
+  // Set only one parameter, clearing others implicitly if underlying SDK works
+  // like that
+  std::map<std::string, firebase::Variant> single_default;
+  single_default["only_this"] = 45.6;
+  firebase::analytics::SetDefaultEventParameters(single_default);
+  firebase::analytics::LogEvent("event_with_only_one_default");
+  LogInfo("Logged event_with_only_one_default");
+
+  // Clear all defaults by passing an empty map.
+  std::map<std::string, firebase::Variant> empty_defaults;
+  firebase::analytics::SetDefaultEventParameters(empty_defaults);
+
+  // Log an event with no defaults.
+  firebase::analytics::LogEvent("event_with_no_defaults");
+  LogInfo("Logged event_with_no_defaults");
+
+  // If we reach here without crashing, consider the basic test passed for the
```

Comment:
As a full smoke test, set a map with every single possible type including sub-map and try that.